lib/AFS/CellCC/Dump.pm

```diff
@@ -41,35 +41,42 @@ our @EXPORT_OK = qw(process_dumps);
 # currently just calculate a checksum for the dump blob, and update the
 # database to give the dump filename and metadata.
 sub
-_dump_success($$$$) {
-    my ($jobid, $dvref, $prev_state, $dump_fh) = @_;
+_dump_success($$$) {
+    my ($prev_state, $jobs_ref, $dump_fhs_ref) = @_;
+    my @jobs = @{$jobs_ref};
+    my @dump_fhs = @{$dump_fhs_ref};
     my $done_state = 'DUMP_DONE';
-    my $filesize = stat($dump_fh)->size;
+    my $filesize = stat($dump_fhs[0])->size;
 
-    my $base_file = basename($dump_fh->filename);
-
-    seek($dump_fh, 0, 0);
+    seek($dump_fhs[0], 0, 0);
 
+    foreach my $i (1 .. $#dump_fhs) {
+        File::Copy::copy($dump_fhs[0]->filename, $dump_fhs[$i]->filename);
```

Comment:
I'm not sure I see a reason to do a full file copy? Just using `link($dump_fhs[0]->filename, $dump_fhs[$i]->filename);` will be much faster, and we don't modify the files after this regardless.